### PR TITLE
Sync utility

### PR DESCRIPTION
**Pull request by peternga**
Self-contained script to fetch username/email from Github given a user's Github API Token. It will get the user's list of emails and find the one matching the domain passed in -- in our case, clever.com. It will then attempt to update the entry matching the email in DynamoDB with the new Github username.

@xavi-

### CODE REVIEW

File: syncself/syncself.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		UpdateItem(dynamoTable, dynago.HashKey("email", email)).
+		UpdateExpression("SET github=:github").
+		Param(":github", username).
+		Execute()
+	if err != nil {
+		log.Fatalf("Failed to update '%s' field. Tried to set value to '%s' for user with email '%s'", email, username)
+	}
+
+	log.Printf("Successfully updated for source github, username %s, email addr %s", username, email)
+}
```
Comment (xavi-) [inline syncself/syncself.go]: Nit. Git diff are a little nicer if there's a new line at the end of logs.  Sublime will automatically add one on save if this setting is enabled:

```
	"ensure_newline_at_eof_on_save": true
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+SHELL := /bin/bash
```
Comment (xavi-) [inline Makefile]: Because this Makefile is syncself specific, it might be best to move it to the syncself directory.

File: syncself/sync.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env sh
+# Script to test locally. To actually run, use binary in salt-dev-env
+
+DYNAMO_TABLE="who-is-who" \
+DYNAMO_REGION="us-west-1" \
+DYNAMO_ENDPOINT="http://localhost:8000" \
+DOMAIN="clever.com" \
+GITHUB_ORG="Clever" \
+go run syncself.go
```
Comment (xavi-) [inline syncself/sync.sh]: It'd be cool if this script could be integrated into a `make run`.

File: syncself/syncself.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	dynamoEndpoint string
+	dynamoRegion   string
+	domain         string
+)
+
+// m is a convenience type for using kayvee.
+type m map[string]interface{}
+
+// requiredEnv tries to find a value in the environment variables. If a value is not
+// found the program will panaic.
```
Comment (xavi-) [inline syncself/syncself.go]: Nit.  s/panaic/panic/

Also, it doesn't look like this function will actually cause a panic.  It just logs a fatal.

File: syncself/syncself.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	opt := &githubAPI.ListOptions{}
+	emails, _, err := gh.Users.ListEmails(opt)
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to form HTTP request for Github => {%s}", err)
+	}
+
+	for _,e := range emails {
+		email := *e.Email
+		emailSplit := strings.Split(email, "@")
+		isClever := domain == emailSplit[len(emailSplit)-1]
```
Comment (xavi-) [inline syncself/syncself.go]: Could replaces with [`strings.HasSuffix(email, domain)`](http://golang.org/pkg/strings/#HasSuffix).

File: syncself/syncself.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/Clever/who-is-who/integrations"
+	githubAPI "github.com/google/go-github/github"
+	"golang.org/x/oauth2"
```
Comment (xavi-) [inline syncself/syncself.go]: If you have a chance and if they're available, can you try to use vendored versions of "github.com/google/go-github/github" and "golang.org/x/oauth2"?

Comment (peternga) [inline syncself/syncself.go]: How can I find the vendored version?

Comment (xavi-) [inline syncself/syncself.go]: If the library is on github and they owner tags releases, you can use use "gopkg.in/<repo-owner>/<repo>.<tag-name>".  If they don't, I'm not sure honestly.

Vendoring the imports is just a nice to have.  It shouldn't be a blocker.

Comment (peternga) [inline syncself/syncself.go]: Ok, it looks like neither of those repos have release versioning so I'll leave as is.

**Pull request closed without merging**